pkg/math/oracle/utils.go

```diff
@@ -4,54 +4,72 @@ import (
 	"fmt"
 	"math/big"
 
-	"github.com/skip-mev/slinky/oracle/types"
-	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
+	slinkytypes "github.com/skip-mev/slinky/pkg/types"
+	mmtypes "github.com/skip-mev/slinky/x/mm2/types"
 )
 
-// GetTickerFromOperation returns the ticker for the given operation.
-func (m *MedianAggregator) GetTickerFromOperation(
-	operation mmtypes.Operation,
+// GetTickerFromCurrencyPair returns the ticker for the given currency pair.
+func (m *MedianAggregator) GetTickerFromCurrencyPair(
+	cp slinkytypes.CurrencyPair,
 ) (mmtypes.Ticker, error) {
 	cfg := m.GetMarketMap()
-	ticker, ok := cfg.Tickers[operation.CurrencyPair.String()]
+	market, ok := cfg.Markets[cp.String()]
 	if !ok {
-		return mmtypes.Ticker{}, fmt.Errorf("missing ticker: %s", operation.CurrencyPair.String())
+		return mmtypes.Ticker{}, fmt.Errorf("missing ticker: %s", cp.String())
 	}
 
-	return ticker, nil
+	return market.Ticker, nil
 }
 
 // GetProviderPrice returns the relevant provider price. Note that if the operation
 // is for the index provider, then the price is retrieved from the previously calculated
 // median prices. Otherwise, the price is retrieved from the provider cache. Additionally,
 // this function normalizes (scales, inverts) the price to maintain the maximum precision.
 func (m *MedianAggregator) GetProviderPrice(
-	operation mmtypes.Operation,
+	ticker mmtypes.Ticker,
+	providerConfig mmtypes.ProviderConfig,
 ) (*big.Int, error) {
-	ticker, err := m.GetTickerFromOperation(operation)
+	cache := m.GetDataByProvider(providerConfig.Name)
+
+	price, ok := cache[ticker]
+	if !ok {
+		return nil, fmt.Errorf("missing %s price for ticker: %s", providerConfig.Name, ticker.String())
+	}
+
+	scaledPrice, err := ScaleUpCurrencyPairPrice(ticker.Decimals, price)
 	if err != nil {
 		return nil, err
 	}
 
-	var cache types.TickerPrices
-	if operation.Provider != mmtypes.IndexPrice {
-		cache = m.GetDataByProvider(operation.Provider)
-	} else {
-		cache = m.GetAggregatedData()
+	if providerConfig.Invert {
+		scaledPrice = InvertCurrencyPairPrice(scaledPrice, ScaledDecimals)
 	}
 
-	price, ok := cache[ticker]
-	if !ok {
-		return nil, fmt.Errorf("missing %s price for ticker: %s", operation.Provider, ticker.String())
+	return scaledPrice, nil
+}
+
+// GetIndexPrice returns the aggregated index price.
+func (m *MedianAggregator) GetIndexPrice(
+	providerConfig mmtypes.ProviderConfig,
+) (*big.Int, error) {
```

Comment:
small nit: I think this should just accept the `NormalizebyPair`

Reply:
We already know that this is only called if normalize by pair is not nil so you can just pass in the de-ref pair here.

Reply:
simplifies the API